packages/next/build/jest/jest.ts

```diff
@@ -157,7 +157,7 @@ export default function nextJest(options: { dir?: string } = {}) {
           // To match Next.js behavior node_modules is not transformed, only `transpiledPackages`
           ...(transpiled
             ? [
-                `/node_modules/(?!(${transpiled}))/`,
+                `/node_modules/(?!(${transpiled})/)`,
```

Comment:
We could also omit the trailing `/` entirely, which would allow partial matching of package names. I think it's best to be explicit though, and seems to align with the intent of the config option.